src/prototype_creep_routing.js

```diff
@@ -162,13 +162,36 @@ Creep.prototype.followPath = function(action) {
     this.memory.routing.reached = true;
     return action(this);
   }
-  return this.moveByPathMy(route, routePos, 'pathStart', this.memory.routing.targetId, false, action);
+  const prepareData = this.moveByPathPrepare(route, routePos, 'pathStart', this.memory.routing.targetId);
+  if (prepareData.unit.preMove) {
+    if (prepareData.unit.preMove(this, prepareData.directions)) {
+      return true;
+    }
+  }
+  return this.moveByPathMy(route, routePos, 'pathStart', this.memory.routing.targetId, action, prepareData);
 };
 
-Creep.prototype.moveByPathMy = function(route, routePos, start, target, skipPreMove, action) {
-  const unit = roles[this.memory.role];
+Creep.prototype.moveByPathPrepare = function(route, routePos, start, target) {
+  const result = {};
+  result.unit = roles[this.memory.role];
```

Comment:
Could it make sense to store `unit` as a default attribute on the Creep instead?